src/DI/BackgroundQueueExtension.php

```diff
@@ -102,8 +103,8 @@ public function afterCompile(ClassType $class)
 	'. $serviceMethod->getBody() .'
 })();
 if (php_sapi_name() === "cli") {
-	$this->getByType(\Symfony\Component\EventDispatcher\EventDispatcherInterface::class)
-		->addListener(\Symfony\Component\Console\ConsoleEvents::TERMINATE, function () use ($service) {
+	$this->getByType(\Kdyby\Events\EventManager::class)
+		->addEventListener(\Symfony\Component\Console\ConsoleEvents::TERMINATE, function () use ($service) {
```

Comment:
@NoxZet proc se delalo toto?

Reply:
@thorewi už si úplně nepamatuji, předpokládám, že mi to s novýma knihovnama nějak přestalo fungovat

Reply:
@NoxZet tak to jeste jednou testni s tim puvodnim a dej mi vedet co tam byl za problem...

Reply:
a testni to na 2.4 i na 3

Reply:
@thorewi chceme Tedy nechceme vůbec používat Kdyby. Problém byl že Kdyby má od nějaké verze nějakou vlastní implementaci EventDispatcherInterface, které pak naimplementovanou metodu addListener, proto to někde kde se používá Kdyby to přestalo fungovat. Zřejmě tedy pokusit se odstranit Kdyby na projektu a tady to vrátit do předchozího stavu.

Reply:
@NoxZet jo zkusme to udelat bez zavislosti na kdyby, to mi dava smysl...

Reply:
@thorewi mělo by to být ready na merge, vydal bych to tedy jako novou major verzi kvůli rozbití zpětné kompatibility v Nette 3

Reply:
ok ale urcite se mi jeste nelibi zavislost na nette 3.0... vyjmenujme konkretni baliky, ktere to potrebuje...

Reply:
a co presne rozbiji tu zpetnou kompatibilitu s nette 2.4?

Reply:
Jak to tak zkoumám, myslím že kód samotný by měl fungovat na obou. V composer.json ale třeba `kdyby/rabbitmq` 1.1/1.2 určitě nebude fungovat s Nette 3 a naopak `adt/rabbitmq` 2.0 nebude fungovat s Nette 2.4.

V Nette 3 se například používají v interfacech metody s otypovaným výstupem, což pak způsobuje problémy, protože pokud se to tak nenapíše do implementace interface, tak nesedí signatura a stejně tak když to je v implementaci navíc. [Viz tady.](https://github.com/peldax/RabbitMq/compare/v1.2.0...AppsDevTeam:master#diff-c68f8b39dc1e83fd3967f4b52fd4fcba)